### DISCUSSION TITLE
persist and hydrate examples

### DISCUSSION THREAD

**Discussion by busslina** [GregoryConrad/rearch-dart#121] [General]
Hi, I didn't find any example or extended info on `persist` and `hydrate` side effects.

Can you provide a minimal example on them. I just can't visualize their utility (but I believe they could be great for me).

Thanks

**Comment by busslina** [comment]
Can them be maybe util for auto handle its internal state based on other capsule states?

**Comment by GregoryConrad** [comment]
`use.persist` is basically a `use.state` that persists the state elsewhere, per its `read` and `write` function parameters. So, if you ever have a `use.state` you want to persist the value of (e.g., to a database), you can probably just switch it to a `use.persist`.

`use.hydrate` is similar to `use.persist` but is designed to work with side effects that manage their own state, like `use.replay`. That allows you to use the state-managing side effect (e.g., `use.replay`) on its own, but then also add a layer of persistence on top of that.

**Reply by busslina** [reply]
I see.. thanks :)
